justifi-tech/web-component-library
# FE-144: Check for contentWindow before attempting postMessage

There have been a number of occurrences where a `postMessage` is attempted but `iframe.contentWindow` is not defined. This could be a race condition with the iframe being rendered in the DOM, or it could be that the `message` event handler is not being properly cleaned up in all situations, or both. In any case, however, we should handle this more gracefully and check that `iframe.contentWindow` is defined before attempting to call `postMessage` on it


Links
-----



Developer considerations
--------------

  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?


Steps for Testing/QA
--------------------
- Pull this branch and run the build from the `stencil-library` directory
- Run `yarn link` from the `stencil-library` (you may already have it linked, that's fine)
- Download this [vue sample app](https://github.com/justifi-tech/web-component-library/files/11401620/vue-sample-app.zip) and unzip it somewhere
- Inside the unzipped vue app directory, run `yarn link "@justifi/webcomponents"`
- Inside the unzipped vue app directory, run `yarn` and then `yarn run dev`
- When the dev server starts, it will say what port it is running on. Open it up in your browser
- The app should consist of the card form component and a button to toggle it (there is a vue template condition around the component)
- With the dev tools open, click the toggle a few times and ensure that there is no error related to `contentWindow` being `null` or `undefined`
